### DISCUSSION TITLE
Where can I get the final model

### DISCUSSION THREAD

**Discussion by thunder-007** [Project-MONAI/tutorials#1470] [Q&A]
I don't have enough compute power to train the model by myself. It would be helpful if I can get the huggingfaces link also the script to load and test a test script with nii.gz file and slice as input and predicts the output https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/brats_segmentation_3d.ipynb

model.load_state_dict(torch.load(os.path.join(root_dir, "best_metric_model.pth")))

**Comment by KumoLiu** [comment]
Hi @Thunder-007, sorry that I didn't fully understand your question.
In the tutorial link you shared, it saved the best model by using:
```
torch.save(
                    model.state_dict(),
                    os.path.join(root_dir, "best_metric_model.pth"),
                )
```
You can also use `torch.save` to save the final model.
Hope it helps, thanks!

**Comment by thunder-007** [comment]
HI @KumoLiu I mean I don''t have infrastructure to train where can I directly find the pretrained model also I'm confused with the inference and preprocessing is there some simple method which takes the .nii.gz file and slice number as input and give the mask and the image as ouput.

**Reply by KumoLiu** [reply]
Hi, @Thunder-007 The model-zoo repository (https://github.com/Project-MONAI/model-zoo/releases/tag/hosting_storage_v1) contains some pre-trained models.
Additionally, there might be some inference transforms because you must ensure that your inputs have the same orientation and spacing as your training data.
Thanks!

**Reply by thunder-007** [reply]
@KumoLiu thanks bro I'm stuck with the preprocessing and loading the model part can I have a example for using the pretrained model like 
```python3
def predictor(3d_file,other_requireparams):
    .....
    .....
    return [tumoreimage(numpyarray),mask(numpy_array)

tumor_image , mask = predictor('3dfile_path',slice_number)
plt.imshow(tumor_image)
plt.imshow(mask)
```

**Reply by KumoLiu** [reply]
Hi @Thunder-007, here is a simple demo.
```
model.load_state_dict(torch.load(os.path.join(root_dir, "best_metric_model.pth")))
model.eval()
with torch.no_grad():
    # select one image to evaluate and visualize the model output
    val_input = val_ds[6]["image"].unsqueeze(0).to(device)
    roi_size = (128, 128, 64)
    sw_batch_size = 4
    val_output = inference(val_input)
    val_output = post_trans(val_output[0])
```
You can also refer to this tutorial.
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/torch/unet_inference_dict.py

Hope it helps, thanks!

**Reply by thunder-007** [reply]
@KumoLiu I don't have the entire dataset when user gives the input I wanna do It how can I preprocess just one file instead of loading it as a whole just the path of one file.
```
path , slice_number  = 'examples/BRATS_485.nii.gz' , 65
```

**Reply by KumoLiu** [reply]
I don't believe the length of the data affects the pipeline; you can only put one data inside the dataset.

**Reply by thunder-007** [reply]
That's what I'm trying to do now but any simpler way will be helpful thanks a lot.

**Reply by thunder-007** [reply]
@KumoLiu It didn't worked the dataset.json is playing some role while pre processing we can't give 1 image as such

**Reply by KumoLiu** [reply]
Hi @Thunder-007, I didn't fully get your question, if you are using a dataset.json, you may also update the json file.
And for what I mean, you can only put one file in this `files`.
https://github.com/Project-MONAI/tutorials/blob/d0de28e3e184847a4adf84bbc8f8f6e8b699c012/3d_segmentation/torch/unet_inference_dict.py#L51C5-L51C10

**Reply by thunder-007** [reply]
@KumoLiu sorry for the confusion I will explain as simple as possible kindly tell me if something isn't understandable
```
import monai
model = monai.bundle.load("brats_mri_segmentation")
images , masks = model.predict("path/of/nii.gz file") # I wanted to get the tumor images and segmentation from the model
```
I'm will get the pretrained model from the bundle.load method how can make inference now use the model to create a segment as numpu array simply with just the single nni.gz
![Screenshot from 2023-07-27 16-10-38](https://github.com/Project-MONAI/tutorials/assets/75431678/695a50cd-7e96-4c2d-8285-b88b93d36051)
I just wanted to make a simple ui for the model.

**Reply by thunder-007** [reply]
thanks a lot for the above reply https://github.com/Project-MONAI/tutorials/blob/d0de28e3e184847a4adf84bbc8f8f6e8b699c012/3d_segmentation/torch/unet_inference_dict.py#L51C5-L51C10 can I have same such inferece.py for SegRestNet similarly?

**Reply by KumoLiu** [reply]
Yes, sure.

**Reply by thunder-007** [reply]
Thanks I'm finally able to achieve it I will make a notebook and a pull request thank you...
